src/Core/src/ImageSources/UriImageSourceService/UriImageSourceService.iOS.cs

```diff
@@ -21,20 +23,35 @@ public partial class UriImageSourceService
 
 			try
 			{
-				// TODO: use a real caching library with the URI
-				if (imageSource is not IStreamImageSource streamImageSource)
-					return null;
+				var hash = Crc64.ComputeHashString(imageSource.Uri.OriginalString);
+				var pathToImageCache = CacheDirectory + hash + ".png";
+
+				NSData? imageData;
+
+				if (IsImageCached(pathToImageCache))
+				{
+					imageData = GetCachedImage(pathToImageCache);
+				}
+				else
+				{
+					// TODO: use a real caching library with the URI
+					if (imageSource is not IStreamImageSource streamImageSource)
+						return null;
 
-				using var stream = await streamImageSource.GetStreamAsync(cancellationToken).ConfigureAwait(false);
+					using var stream = await streamImageSource.GetStreamAsync(cancellationToken).ConfigureAwait(false);
 
-				if (stream == null)
-					throw new InvalidOperationException($"Unable to load image stream from URI '{imageSource.Uri}'.");
+					if (stream == null)
+						throw new InvalidOperationException($"Unable to load image stream from URI '{imageSource.Uri}'.");
 
-				var data = NSData.FromStream(stream);
-				if (data == null)
-					throw new InvalidOperationException("Unable to load image stream data.");
+					imageData = NSData.FromStream(stream);
+
+					if (imageData == null)
+						throw new InvalidOperationException("Unable to load image stream data.");
+
+					await CacheImage(imageData, pathToImageCache);
```

Comment:
Can you put a .ConfigureAwait(false) on the end of this call?

Reply:
removed the Task / await stuff since it's not needed